imap_processing/swe/tests/test_swe_l1b.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.swe.l0 import decom_swe
+from imap_processing.swe.l1a.swe_l1a import create_dataset, group_by_apid
+from imap_processing.swe.l1a.swe_science import swe_science
+from imap_processing.swe.l1b.swe_l1b import convert_raw_to_eu
+
+
+@pytest.fixture(scope="session")
+def decom_test_data():
+    """Read test data from test folder"""
+    test_folder_path = "swe/tests/l0_data"
+    packet_files = list(imap_module_directory.glob(f"{test_folder_path}/*.bin"))
+
+    data_list = []
+    for packet_file in packet_files:
+        data_list.extend(decom_swe.decom_packets(packet_file))
+    return data_list
+
+
+def test_swe_l1b(decom_test_data):
+    """Test that calculate engineering unit(EU) is
+    matches validation data.
+
+    Parameters
+    ----------
+    decom_test_data : _type_
+        _description_
+    """
+    # SWE_APP_HK = 1330
+    # SWE_SCIENCE = 1344
+    grouped_data = group_by_apid(decom_test_data)
+    # Process science to l1a.
+    # because of test data being in the wrong
+    # order, we need to manually re-sort data
+    # into order.
+    sorted_packets = sorted(
+        grouped_data[1344], key=lambda x: x.data["QUARTER_CYCLE"].raw_value
+    )
+    science_l1a_ds = swe_science(sorted_packets)
+    science_l1b = convert_raw_to_eu(science_l1a_ds)
+
+    # read science validation data
+    test_data_path = f"{imap_module_directory}/swe/tests/l0_validation_data"
+    eu_validation_data = pd.read_csv(
+        f"{test_data_path}/idle_export_eu.SWE_SCIENCE_20230927_172708.csv",
+        index_col="SHCOARSE",
+    )
+    second_data = sorted_packets[1]
+    validation_data = eu_validation_data.loc[second_data.data["SHCOARSE"].raw_value]
+
+    science_eu_field_list = [
+        "SPIN_PHASE",
+        "SPIN_PERIOD",
+        "THRESHOLD_DAC",
+    ]
+
+    # Test EU values for science data
+    for field in science_eu_field_list:
+        assert round(science_l1b[field].data[1], 5) == round(validation_data[field], 5)
```

Comment:
Would we be able to use numpys testing routines for this at all instead of rounding things yourself, and possibly removing the loop too then.
https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose

Reply:
btw, I looked into this few days ago. I couldn't get it to work without complicating logics to remove the loop. The reason was that, I am only check float value matches for only these three variables. If we remember correctly, I had to look up these values from both `science_l1b` and `validation_data`, then match them using `np`.
```
science_eu_field_list = [
        "SPIN_PHASE",
        "SPIN_PERIOD",
        "THRESHOLD_DAC",
    ]
```